server/src/db/queries/user/updateStepStatus.js

```diff
@@ -0,0 +1,28 @@
+const Query = require("../../core/Query")
+
+const updateStepStatus = ({ username, stepId, status }) => {
+    return new Query({
+        statement: `
+        MATCH (u:User {username: $username}), (s:Step {id: $stepId})
+        OPTIONAL MATCH (u)-[prev]->(s)
+        DELETE prev
+        WITH u, s
+        CALL apoc.do.case([
+            $status =~ "HAS_COMPLETED", 
```

Comment:
Why create edges of different types? I'd suggest create edges of the same type with different properties instead.

Reply:
`neo4j-graphql` doesn’t support properties in relationships. Had to create different types of relations because of this.

Reply:
Does this not work - https://grandstack.io/docs/graphql-relationship-types/#relationships-with-properties